### ISSUE TITLE
A way to get SCSS ?

### ISSUE THREAD

**Issue by thirdman** [antonybudianto/cra-universal#192]
Thanks for this project! I see from an earlier (closed) issue that scss is not supported. Feel free to ignore this one if it's not trivial. 🤣 

TLDR: 
I thought I'd try to customise webpack to use node-sass so that I can use scss. Would this work? Or are you doing other things stopping support for SCSS?

---
Longer version: 

I have a pre-existing project that I was looking to convert SSR. Since I didn't want to re-write all the SCSS and since CRA supports scss I tried to customise the webpack config. Here's what I added to crau.config.js:

```
module: {
        rules: [
          {
            test: /\.s[ac]ss$/i,
            use: [
              // Creates `style` nodes from JS strings
              "style-loader",
              // Translates CSS into CommonJS
              "css-loader",
              // Compiles Sass to CSS
              "sass-loader"
            ]
          }
        ]
      
```
When that choked on .js files, I included the js test and used the babel presets from `/node_modules/crau-universal/src/index.js`:

```
{
            test: /\.js$/,
            exclude: /node_modules/,
            use: {
              loader: "babel-loader",
              options: {
                presets: ["@babel/preset-env"],
                plugins: [require("@babel/plugin-proposal-object-rest-spread")]
              }
            }
          },
```

but this gets me the following error:

```
Error: Module build failed (from ./node_modules/cra-universal/node_modules/babel-loader/lib/index.js):
server/app.js: Unexpected token (10:33)

   8 | const app = createReactAppExpress({
   9 |   clientBuildPath,
> 10 |   universalRender: (req, res) => <App />
            
```

At which point, nearing the limit of things I'm prepared to mess with, I decided to reconsider. 😄 

So I guess my question is, is this approach likely to work? What am I missing?  lol.

**Comment by antonybudianto** [comment]
missed preset-react

**Comment by thirdman** [comment]
argh! Thanks! 😄 I appreciate your time. 🎁 💝 😍 

After wading through various (unrelated, and mostly react-dnd) webpack errors, I believe I have it working.

crau.config.js webpack override uses: 

```
module: {
        rules: [
          {
            test: /\.js$/,
            exclude: /node_modules/,
            use: {
              loader: "babel-loader",
              options: {
                presets: [                  
                    "@babel/preset-env",
                  "@babel/preset-react"
                ],
                plugins: [
                  require("@babel/plugin-proposal-object-rest-spread"),
                  ["@babel/plugin-proposal-class-properties"],
                ]
              }
            }
          },
          {
            test: /\.s[ac]ss$/i,
            use: [
              // Translates CSS into CommonJS
              "css-loader",
              // Compiles Sass to CSS
              "sass-loader"
            ]
          },
          {
            test: /\.svg/,
            use: {
              loader: "svg-url-loader",
              options: {}
            }
          }
        ]
      }

```

**Comment by antonybudianto** [comment]
Nice, I will put this in doc soon!

**Comment by komachi** [comment]
This is closed issue, but searching brought this, and docs should probably cover this case, so if anyone looking for some little hack, here it is (`crau.config.js`):

```
module.exports = {
  modifyWebpack: config => {
    return {
      ...config,
      module: {
        ...config.module,
        rules: [
          ...config.module.rules,
          {
            test: /\.s[ac]ss$/i,
            use: [
              'isomorphic-style-loader',
              'css-loader',
              'sass-loader'
            ]
          }
        ]
      }
    };
  }
};
```

Works perfectly.
